### PR TITLE
Rework the way we find the grid in the page to fix EE behats

### PR DESCRIPTION
**Pull request by jjanvier**

### CODE REVIEW

**Comment by solivier** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Comment by skeleton** [comment]
gtm

**Comment by BitOne** [comment]
GTM

File: features/Context/Page/Base/Grid.php [hunk 1]
```diff
@@ -58,16 +58,12 @@ public function getGrid()
...
         return $this->spin(
             function () {
-                $grids = $this->getElement('Container')->findAll('css', $this->elements['Grid']['css']) +
-                    $this->getElement('Dialog')->findAll('css', $this->elements['Grid']['css']);
-
-                foreach ($grids as $grid) {
-                    if ($grid->isVisible()) {
-                        return $grid;
-                    }
+                $modal = $this->getElement('Body')->find('css', $this->elements['Dialog']['css']);
```
Comment (jmleroux) [inline features/Context/Page/Base/Grid.php]: what is the difference with a `$this->find`
the `getElement` is used to restrict a search in a DOM node, but i'm not sure it is very useful when this node is 'body'

**Comment by jmleroux** [comment]
GTM once answered

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -58,16 +58,12 @@ public function getGrid()
...
+                if (null !== $modal && $modal->isVisible()) {
+                    return $modal->find('css', $this->elements['Grid']['css']);
                 }
 
-                return false;
+                return $this->getElement('Container')->find('css', $this->elements['Grid']['css']);
```
Comment (juliensnz) [inline features/Context/Page/Base/Grid.php]: what if there are multiple grid on the page ?

Comment (juliensnz) [inline features/Context/Page/Base/Grid.php]: Maybe there is no cases where we have multiple grid in the page except one in the modal and one on the background

File: features/Context/Page/Base/Grid.php (hunk 1 above)
Comment (jjanvier) [inline features/Context/Page/Base/Grid.php]: $this->find just doesn't work
I don't know why but you get an exception that says "you must first visit the page"
it's about the Sensio extension

File: features/Context/Page/Base/Grid.php (hunk 2 above)
Comment (jjanvier) [inline features/Context/Page/Base/Grid.php]: it seems there is no case as the behat pass

**Pull request merged**